### DISCUSSION TITLE
Dr. Nash's Review (External Supervisor)

### DISCUSSION THREAD

**Discussion by kimjoyc** [janash/MolecularNodes#15] [Ideas]
Dr. Jessica Nash 
Option 1: The user inputs the solvent and shell information in when the trajectory is loaded.
Information at load time: MD topology + trajectory. Solvent, solute selection; shell selection; frame number.
Advantages: This is the easiest route to go.
Disadvantages: This route lacks flexibility and will require reloading the trajectory into a new blender object every time a new solvation shell or frame needs to be visualized.
Option 2: The user loads an MD trajectory with solvent information and can dynamically select the shell and frame.
Information at load time: MD topology + trajectory. Solvent, solute selection.
Advantages: The most flexible and probably the solution many would expect.
Disadvantages: Unclear if possible. Might require altering a lot of the MolecularNodes package .
We could get Option 1 working as a minimum viable product (MVP) and move on to trying to implement Option 2.

Short summary on how the feedback you received would impact your project moving forward 
Dr. Nash’s feedback includes implementation options for the project overall. The first option entails a UI input with selection panels for the structural specifications and the second option includes a more complex implementation strategy that will render the software more flexible, but will be significantly harder to achieve. The first implementation to use best aligns with the proposed functionality, but lacks flexibility and requires reloading the trajectory into a new blender object every time a new solvation shell or frame needs to be visualized. Moving forward in reporting project updates, we will include the first option designated to be the implementation of choice, and will constitute the MVP for this project, while option two will be a “reach goal.” In this way, we can make sure that our project goals will be met on time by prioritizing development of the MVP.

Reviewers’ perception and clarification
Dr. Nash’s review states that there will be several implementations to fulfill the aim of creating a molecular solvation visualization functionality in MolecularNodes. The first option being the implementation outlined in the capstone proposal, where selection features will be added to the UI. The second option is more open-ended in that there are several ways to accomplish the molecular solvation visualization feature that may be more dynamic than the former approach. The second option would constitute something of a “power user” implementation, allowing more sophisticated users to take advantage of a more flexible selection system. However, the primary focus of the project remains user friendliness and accessibility, and this aspect will continue to be prioritized. 

Project Descriptions and Deliverables are well understood by others
The project descriptions of the aim of the project being a user-friendly molecular solvation visualization functionality added to MolecularNodes is a well understood objective. This is illustrated through Dr. Nash's option 1 guidelines. The final deliverable is well understood, as Dr. Nash states the first option will accomplish our goal to visualize the solute-solvent interactions. 

Questions
Which implementation will be employed and if these are accurate to the proposal?
We chose to prioritize option 1, due to its user-friendly nature. Option 1 also sticks closely to what was proposed in the capstone project, ensuring that all proposed features are included in the final deliverable. 

Concerns (including risks)
Dr. Nash’s review seems to be largely focused on defining a strategy for achieving a MVP within the remaining timeframe of this capstone project. According to this review, we have a clear strategy for creating a deliverable that meets the objectives listed in the capstone proposal. In organizing the remainder of this project, Dr. Nash suggests following a workflow where we concentrate on developing a UI-based implementation of SolvationAnalysis. When this objective is met, any remaining time and effort can be spent on improving the flexibility and power of this package, but only once the useability improvements outlined in the capstone proposal are met.

Achieving objective 2 will eliminate the need to reload another Blender structure with the same structure but with a different set of selections. However, this goal was not highlighted in the project proposal, so it will be pursued only if time permits. 

Recommendations 
The recommendation was to go with option 1 to build a selection function that relies on information from solutes and solvent groups selection. This best aligns with the object of the project to establish a molecular solvation visualization function in MolecularNodes.

Working in this fashion (prioritizing the key goals in option 1, and only then working on the “reach goals” in project 2) will allow us to develop a package that is as flexible and powerful as possible while avoiding Scope Creep.

**Comment by janash** [comment]
Mention progress being made on the BugFix. It could help with the MVP implementation
